doc/GettingStarted.md

```diff
@@ -0,0 +1,225 @@
+# Getting Started
+
+On this page you will learn about [TSL](#TSL) and its Generator, what it is inteded for and how to utilize it.
+
+
+# Introduction
+
+### **What is TSL and what is it used for?**
+
+TSL is a C++ Header-only Library which abstracts elemental operations on 
+hardware specific instructions sets, espacial SIMD / vector operations.
+
+### **Why the indirection through a generator? Why no plain header only library?**
+
+While developing TSL we found, that a large potion is redundant boiler plate code. 
+When looking into the generated files, this should be clear.
+So to avoid writing the same boiler plate over and over again, we implemented the generator.
+Also the generator automatically detects the CPU flags set on your system, 
+so you don't have to specify each vector extension flag yourself.
+
+### **What is the target plattform?**
+
+Tested on Linux (gcc , clang) & WSL
+
+# Setup
+
+
+### 1 - Getting the repository
+
+Clone the git repositiory of the [TSL Generator](https://github.com/db-tu-dresden/TVLGen) (or one of the forks) using the recursive flag.
+
+~~~bash
+git clone --recurse-submodules https://github.com/db-tu-dresden/TVLGen <clone directory>
+~~~
+
+ ### 2 - Install dependencies
+
+ For this project you need to have cmake (Version 3.13 or above) and python (Version 3.10.* or above) installed.
+
+ There are some python dependencies, which are needed to run the generator.
+ Install them using pip:
+
+~~~bash
+pip3 install -r requirements.txt
+~~~
+
+The package pygraphviz also depends on graphviz.
+So make sure to install graphviz as well.
+
+<!-- Todo: requirements installieren, vorab graphviz-dev installieren (sudo apt isntall graphviz-dev) -->
+
+### 3 - Run the generator
+
+As already mentioned the generator will create the TSL library.
+By default, the output is directed into the generator dir, but you can specifiy an output path like following:
+
+~~~bash
+cd <clone directory>
+cmake . -D GENERATOR_OUTPUT_PATH=<path to output directory> -B build/
+~~~
+
+The CMake system is used to detect system specific hardware properties (i.e. which cpu flags are available) and hand them over to the generator.
```

Comment:
Actually, py-cpuinfo.
We may should include the pure bash solution using lscpu since at some time we need to query lscpi anyway, and AFAIK, there is no Python package for that